### ISSUE TITLE
Xdebug v3.0 causes JIT to be disabled

### ISSUE THREAD

**Issue by lukasneubauer** [xdebug/xdebug.org#112]
When you run PHP v8.0 with Xdebug v3.0 enabled, JIT is not working regardless that Opcache is on. It seems that Xdebug is overriding `zend_execute_ex()` function and this seems to be the cause of the issue. The error message is following:
```
<br />
<b>Warning</b>:  JIT is incompatible with third party extensions that override zend_execute_ex(). JIT disabled. in <b>Unknown</b> on line <b>0</b><br />
[15-Sep-2021 12:00:00] NOTICE: fpm is running, pid 1
[15-Sep-2021 12:00:00] NOTICE: ready to handle connections
```
When Xdebug if off, both Opcache and JIT work fine.

I have also created repository, where you can easily reproduce the issue. Just follow the readme.

Repository is here: https://github.com/lukasneubauer/docker-php-circular-jit-dependency-dropped.

**Comment by derickr** [comment]
- This is the repository for the Web site, and not for Xdebug.
- This is expected behaviour

**Comment by lukasneubauer** [comment]
Sorry for putting this issue in wrong repository. But how is this expected behavior? Would it be a bad idea to have JIT enabled when using Xdebug? Can you give me more information please? Or link me and article where this is explained?

**Comment by lukasneubauer** [comment]
I found it being mention of Xdebug website https://xdebug.org/docs/compat.
```
Xdebug does not work with PHP's JIT engine, which is implemented in Opcache. When Xdebug is loaded with PHP's JIT on, you will get a warning, and JIT will be disabled.
```
But why is that?

**Comment by krakjoe** [comment]
Overriding the executor function is as old a trick (formal API) as I can remember for the implementation of instrumentation like xdebug, pcov, phpdbg ... 

It is the JIT (part of opcache) that does not support a custom executor, not Xdebug, or any extension that raises the same message. It doesn't support it because it's not practical to do so.

Xdebug is a userland debugger, in other words, a debugger for the virtual machine that for the last 25 and some odd years has executed PHP. There is no reason to expect Xdebug (or any userland, vm debugger) to support machine code. That describes a very different, vastly more complicated kind of project - you can count the number of machine code debuggers (that get used) in the entire universe on one hand.

Before you can call something a debugger, it must at least support breakpoints. In a virtual machine that is simple since we have control over the executor. For machine code that is not so very simple, it involves disassembling code in memory and editing it to  replace the instruction at the start of the desired sequence with a special interrupt (3) instruction.

That I have explained this in three sentences should not lead you to believe this is the kind of work anyone wants to do - remember the number of debuggers in use ...

TL;DR this is very far from happening, ever ...

**Comment by lukasneubauer** [comment]
Thank you very much for your explanation. I get it now. :)
